src/modules/bot/bot.service.ts

```diff
@@ -37,6 +37,27 @@ export class BotService {
     this.logger = new Logger(BotService.name);
   }
 
+  // Reschedule all the notifications, if service restarted.
+  async onModuleInit() {
+    const pendingSchedules = await this.prisma.schedules.findMany({
+      where: {
+        scheduledAt: {
+          gte: new Date(),
+        }
+      }
+    });
+    this.logger.log(`Found ${pendingSchedules.length} pending schedules.`);
```

Comment:
What is the schedule is already set? Would the notifications would be sent twice? Or would it always be wiped clean?

Reply:
`onModuleInit` will always be called only once per service initialization. So this will only be called if the service is restarted. The database will also only upsert the schedule and not add another entry.